components/supervisor/pkg/supervisor/services.go

```diff
@@ -1094,15 +1088,26 @@ func (s *taskService) ListenToOutput(req *api.ListenToOutputRequest, srv api.Tas
 		return nil
 	}
 
-	done := make(chan struct{})
-
 	sub := s.tasksManager.Subscribe()
```

Comment:
What happens when the task closed between this registration and the call to `getTaskStatus` at the beginning of the method?